fastserde/avro-fastserde-tests-common/src/test/java/com/linkedin/avro/fastserde/logical/types/LogicalTypesTestBase.java

```diff
@@ -0,0 +1,214 @@
+package com.linkedin.avro.fastserde.logical.types;
+
+import static com.linkedin.avro.fastserde.FastSerdeTestsSupport.getCodeGenDirectory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.ByteBuffer;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.apache.avro.Conversion;
+import org.apache.avro.Conversions;
+import org.apache.avro.Schema;
+import org.apache.avro.data.TimeConversions;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.specific.SpecificData;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.avro.specific.SpecificRecordBase;
+import org.testng.Assert;
+import org.testng.collections.Lists;
+
+import com.linkedin.avro.fastserde.FastDeserializer;
+import com.linkedin.avro.fastserde.FastGenericDeserializerGenerator;
+import com.linkedin.avro.fastserde.FastGenericSerializerGenerator;
+import com.linkedin.avro.fastserde.FastSerializer;
+import com.linkedin.avro.fastserde.FastSpecificDeserializerGenerator;
+import com.linkedin.avro.fastserde.FastSpecificSerializerGenerator;
+import com.linkedin.avroutil1.compatibility.AvroCompatibilityHelperCommon;
+import com.linkedin.avroutil1.compatibility.AvroVersion;
+
+public abstract class LogicalTypesTestBase {
+
+    private final File classesDir;
+    private final ClassLoader classLoader;
+
+    LogicalTypesTestBase() {
+        try {
+            classesDir = getCodeGenDirectory();
+            classLoader = URLClassLoader.newInstance(new URL[]{classesDir.toURI().toURL()}, getClass().getClassLoader());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected <T extends GenericData> T copyConversions(SpecificData fromSpecificData, T toModelData) {
+        Optional.ofNullable(fromSpecificData.getConversions())
+                .orElse(Collections.emptyList())
+                .forEach(toModelData::addLogicalTypeConversion);
+
+        fixConversionsIfAvro19(toModelData);
+
+        return toModelData;
+    }
+
+    protected <T extends GenericData> void fixConversionsIfAvro19(T modelData) {
+        if (AvroCompatibilityHelperCommon.getRuntimeAvroVersion() == AvroVersion.AVRO_1_9) {
```

Comment:
OK, I revisited this method and `new TimeConversions.TimeMillisConversion()` is not needed (will be removed in fix-commit).

Other 3 conversions
```
new Conversions.DecimalConversion()
new TimeConversions.TimeMicrosConversion()
new TimeConversions.TimestampMicrosConversion()
```
must be manually added because they're missing. They're actually "lost" when downgrading avro from 1.10.2 to 1.10.1:
- https://github.com/krisso-rtb/avro-logical-types-generated-classes/commit/e271f7d9a1ea60fca325a3c0b9260e5068e168bd
but in avro-util/fastserde we observe the difference between `1.10.2` and `1.9.2`.